src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -0,0 +1,621 @@
+        //CHECK IN PLACE TO RESTRICT INITIAL IMPLEMENTATION TO ONLY DV VENDORS.
+        if (StringUtils.equals(determineKfsVendorTypeCodeBasedOnPmwVendorType(pmwVendor.getVendorType()), VendorConstants.VendorTypes.PURCHASE_ORDER)) {
+            KfsVendorDataWrapper kfsVendorDataWrapper = new KfsVendorDataWrapper();
+            kfsVendorDataWrapper.getErrorMessages().add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.PO_VENDOR_PROCESSING_NOT_AUTOMATIC_YET);
+            kfsVendorDataWrapper.setVendorDetail(null);
+            return kfsVendorDataWrapper;
+        }
+        
+        KfsVendorDataWrapper kfsVendorDataWrapper = populateTaxRuleDependentAttributes(pmwVendor, paymentWorksIsoToFipsCountryMap);
+        if (ObjectUtils.isNotNull(kfsVendorDataWrapper.getVendorDetail())) {
+            kfsVendorDataWrapper.getVendorDetail().getVendorHeader().setVendorTypeCode(determineKfsVendorTypeCodeBasedOnPmwVendorType(pmwVendor.getVendorType()));
+            kfsVendorDataWrapper.getVendorDetail().getVendorHeader().setVendorSupplierDiversities(buildVendorDiversities(pmwVendor, paymentWorksToKfsDiversityMap));
+            kfsVendorDataWrapper.getVendorDetail().setVendorDunsNumber(pmwVendor.getRequestingCompanyDuns());
+            kfsVendorDataWrapper.getVendorDetail().setVendorCreditCardIndicator(new Boolean(pmwVendor.isAcceptCreditCards()));
+            kfsVendorDataWrapper.getVendorDetail().setActiveIndicator(true);
+            kfsVendorDataWrapper.getVendorDetail().setVendorUrlAddress(pmwVendor.getRequestingCompanyUrl());
+            kfsVendorDataWrapper.getVendorDetail().setVendorAddresses(buildVendorAddresses(pmwVendor, paymentWorksIsoToFipsCountryMap));
+            kfsVendorDataWrapper.getVendorDetail().setVendorContacts(buildVendorContacts(pmwVendor));
+            kfsVendorDataWrapper.getVendorDetail().setExtension(buildVendorDetailExtension(pmwVendor));
+            kfsVendorDataWrapper.getVendorDetail().setVendorParentIndicator(true);
+            kfsVendorDataWrapper = buildRemainingVendorNotes(pmwVendor, kfsVendorDataWrapper);
+            kfsVendorDataWrapper.getVendorDetail().setBoNotes(kfsVendorDataWrapper.getVendorNotes());
+        }
+        return kfsVendorDataWrapper;
+    }
+    
+    private KfsVendorDataWrapper populateTaxRuleDependentAttributes(PaymentWorksVendor pmwVendor, Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
+        KfsVendorDataWrapper kfsVendorDataWrapper = new KfsVendorDataWrapper();
+        VendorHeader vendorHeader = new VendorHeader();
+
+        switch (determineTaxRuleToUseForDataPopulation(pmwVendor, convertIsoCountryCodeToFipsCountryCode(pmwVendor.getRequestingCompanyTaxCountry(), paymentWorksIsoToFipsCountryMap))) {
+            case PaymentWorksConstants.PaymentWorksNewVendorTaxBusinessRule.INDIVIDUAL_US_SSN: 
+            {
+                vendorHeader.setVendorForeignIndicator(new Boolean(false));
```

Comment:
the code for each of these cases should be extracted in separate methods

Reply:
Separate methods created as requested.